default/AI/ResearchAI.py

```diff
@@ -20,19 +22,396 @@ def get_research_index():
+            unlocked_parts.append(item.name)
+        elif item.type == fo.unlockableItemType.shipHull:
+            unlocked_hulls.append(item.name)
+    if not (unlocked_parts or unlocked_hulls):
+        return 0
+    old_designs = ship_designer.optimize_design(consider_fleet_count=False)
+    new_designs = ship_designer.optimize_design(additional_hulls=unlocked_hulls, additional_parts=unlocked_parts, consider_fleet_count=False)
+    if not (old_designs and new_designs):
+        # AI is likely defeated; don't bother with logging error message
+        return 0
+    old_rating, old_pid, old_design_id, old_cost = old_designs[0]
+    old_design = fo.getShipDesign(old_design_id)
+    old_rating = old_rating
+    new_rating, new_pid, new_design_id, new_cost = new_designs[0]
+    new_design = fo.getShipDesign(new_design_id)
+    new_rating = new_rating
+    if new_rating > old_rating:
+        ratio = (new_rating - old_rating) / (new_rating + old_rating)
+        return ratio * 1.5 + 0.5
+    else:
+        return 0
+
+def get_defense_priority(rng):
+    if foAI.foAIstate.aggression <= fo.aggression.cautious:
+        print "AI is cautious. Increasing priority for defense techs."
+        return 2
+    if foAI.foAIstate.misc.get('enemies_sighted', {}):
+        return 1
+    else:
+        return 0.2
+
+def get_production_boost_priority(rng):
+    return 1.5
+
+def get_research_boost_priority(rng):
+    return 2
+
+def get_production_and_research_boost_priority(rng):
+    return 2.5
+
+def get_population_boost_priority(rng):
+    return 2
+
+def get_supply_boost_priority(rng):
+    # TODO consider starlane density and planet density
+    return 1
+
+def get_meter_change_boost_priority(rng):
+    return 1
+
+def get_detection_priority(rng):
+    # TODO consider stealth of enemies
+    return 1
+
+def get_weapon_priority(rng):
+    if foAI.foAIstate.misc.get('enemies_sighted', {}):
+        return 1
+    else:
+        return 0.1
+
+def get_armor_priority(rng):
+    if foAI.foAIstate.misc.get('enemies_sighted', {}):
+        return 1
+    else:
+        return 0.1
+
+def get_shield_priority(rng):
+    if foAI.foAIstate.misc.get('enemies_sighted', {}):
+        return 1
+    else:
+        return 0.1
+
+def get_engine_priority(rng):
+    return 1 if rng.random() < 0.7 else 0
+
+def get_fuel_priority(rng):
+    return 1 if rng.random() < 0.7 else 0
+
+def get_troop_pod_priority(rng):
+    if foAI.foAIstate.misc.get('enemies_sighted', {}):
+        return 1
+    else:
+        return 0
+
+def get_colony_pod_priority(rng):
+    return 1
+
+def get_stealth_priority(rng):
+    max_stealth_species = get_max_stealth_species()
+    if max_stealth_species[1] > 0:
+        print "Has a stealthy species %s. Increase stealth tech priority" % max_stealth_species[0]
+        return 2.5
+    else:
+        return 0
+
+def get_genome_bank_priority(rng):
+    # TODO boost genome bank if enemy is using bioterror
+    return 1
```

Comment:
This is definitely too high. Turn 1 priorities:

GRO_GENETIC_MED 0.187500    32.00 ['GRO_GENETIC_ENG']
PRO_ROBOTIC_PROD 0.375000    24.00 []
GRO_SUBTER_HAB 0.375000    32.00 ['GRO_PLANET_ECOL']
LRN_ALGO_ELEGANCE 0.500000    24.00 []

I do not think that researching it this early is a good way for the AI. I suggest something like 0.2, maybe slightly higher if we play with high monster setting?